Client/WarFare/UIImageTooltipDlg.cpp

```diff
@@ -1052,6 +1053,79 @@ int	CUIImageTooltipDlg::CalcTooltipStringNumAndWrite(__IconItemSkill* spItem, bo
 	}
 	ERROR_EXCEPTION
 
+	if (spItem->pItemBasic != nullptr)
+	{
+		std::string szRemark = spItem->pItemBasic->szRemark;
+	
+		int strSize = spItem->pItemBasic->szRemark.size();
+		if (strSize > 0)
+		{
+			int splitPos = 0;
+			for (int i = 0; i < strSize; i++)
+			{
+				if (szRemark[i] == '(' || (szRemark[i] == ' ' && i > strSize / 2))
+				{
+					splitPos = i;
+					break;
+				}
+			}
+			if (splitPos > 0)
+			{
+				m_pStr[iIndex]->SetColor(m_CWhite);
+				m_pstdstr[iIndex] = fmt::format("*{}", szRemark.substr(0, splitPos));
```

Comment:
I wouldn't normally mention it, but item tooltips are an unfortunate hotspot for frame drops. This is convenient but also really inefficient, because we're (ignoring the fact that szRemark is copied to begin with) allocating a new string, copying it, then allocating and building a new string from it.

Probably the simplest way to improve this is to change the `substr()` to refer to a `string_view` of the original string.
Something like:
```cpp
m_pstdstr[iIndex] = fmt::format("*{}", std::string_view(szRemark.data(), splitPos));
```
It's slightly more verbose but it's close enough to the original.

This applies to the other case as well.